addons/account/models/company.py

```diff
@@ -496,8 +499,8 @@ def build_move_info(move):
                 results_by_journal['results'].append(rslt)
                 continue
 
-            all_moves_count = self.env['account.move'].search_count([('state', '=', 'posted'), ('journal_id', '=', journal.id)])
-            moves = self.env['account.move'].search([('state', '=', 'posted'), ('journal_id', '=', journal.id),
+            all_moves_count = self.env['account.move'].sudo().search_count([('state', '=', 'posted'), ('journal_id', '=', journal.id)])
+            moves = self.env['account.move'].sudo().search([('state', '=', 'posted'), ('journal_id', '=', journal.id),
```

Comment:
It might be interesting to add a comment explaining why a `sudo()` is necessary, and why it is not an issue since we are only using the search result for hash integrity purposes and not to return the actual values.